core/components/atoms/button/Button.tsx

```diff
@@ -178,7 +178,7 @@ const ButtonElement = React.forwardRef<HTMLButtonElement, ButtonProps>((props, r
                 data-test="DesignSystem-Button--Icon"
                 name={icon}
                 type={iconType}
-                size={largeIcon && !children ? sizeMapping[size] + 4 : sizeMapping[size]}
+                size={size === 'tiny' ? 14 : largeIcon && !children ? sizeMapping[size] + 4 : sizeMapping[size]}
```

Comment:
Store this icon size calculation logic into some variable and then use that variable here